docker_cli/_mount.py

```diff
@@ -0,0 +1,48 @@
+import os
+from typing import Union
+
+
+class Mount:
+    """A description of a docker mount."""
+
+    def __init__(
+        self,
+        host_mount_point: Union[os.PathLike[str], str],
+        image_mount_point: Union[os.PathLike[str], str],
+        permissions: str
+    ):
+        """
+        A description of a docker mount.
+
+        Parameters
+        ----------
+        host_mount_point : os.PathLike[str]
+            The location of the mount in the host environment,
+        image_mount_point : os.PathLike[str]
+            The location of the mount on the image,
+        permissions : str
+            "ro" for readonly, "rw" for read/write.
+
+        Raises
+        ------
+        ValueError
+            _description_
+        """
+        if permissions not in ["ro", "rw"]:
+            raise ValueError(
+                "Permissions value for mount not given as \"ro\" or \"rw\"."
+            )
+        self.permissions = permissions
+        self.host_mount_point = host_mount_point
```

Comment:
I'm generally thinking of handling those cases outside of the BindMount class itself, since it seems like whether or not that is expected would be something for the specific function generating a bind mount would decide. In a lot of cases later down the line in `data fetch` and `workflows`, the system will generate files for the user if they aren't already present (e.g. if the user specifies an `outputs` directory and doesn't already have it created, the code will generate that directory and any necessary subdirectories automatically.)

I guess it would still be possible to check at that time and make sure the mounted files or directories are generated before being named, but in practice that would just demand that the BindMount be instantiated only after those files have been generated.

Reply:
> I'm generally thinking of handling those cases outside of the BindMount class itself

OK, sounds good. In hindsight, I agree that's probably a better approach than my suggestion.